src/plugins/replyPingControl/index.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2023 Vendicated, MrDiamond, ant0n, and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { MessageStore, UserStore } from "@webpack/common";
+
+export const settings = definePluginSettings({
+    alwaysPingOnReply: {
+        type: OptionType.BOOLEAN,
+        description: "Always get pinged when someone replies to your messages",
+        default: false,
+    }
+});
+
+export default definePlugin({
+    name: "ReplyPingControl",
+    description: "Control whether to always or never get pinged on message replies",
+    authors: [Devs.ant0n, Devs.MrDiamond],
+    settings,
+
+    patches: [
+        {
+            find: "_channelMessages",
+            replacement: {
+                match: /receiveMessage\((\i)\)\{/,
+                replace: "$&$self.modifyMentions($1);"
+            }
+        }
+    ],
+
+    modifyMentions(message) {
```

Comment:
import MessageJSON too
```suggestion
    modifyMentions(message: MessageJSON) {
```